relay/relay.go

```diff
@@ -262,15 +270,16 @@ func (r *Relay) addPeer(peerID core.PeerID) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	// TODO: account for peers that rejoin
 	if len(r.peers) == 0 || r.peers[len(r.peers)-1] < peerID {
 		r.peers = append(r.peers, peerID)
 		return
 	}
 
 	// TODO: can optimize via binary search
 	for index, peer := range r.peers {
-		if peer > peerID {
+		if peer == peerID {
```

Comment:
in order to remove the ugly if...else if, L280-L287 could have been another function

Reply:
Refactored using a switch statement, is a little more cleaner, not sure that extracting a function will help here.